hazelcast-client/src/main/java/com/hazelcast/client/config/ClientClasspathYamlConfig.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * A {@link ClientConfig} which is initialized by loading an YAML configuration file from the classpath.
+ */
+public class ClientClasspathYamlConfig extends ClientConfig {
+    private static final ILogger LOGGER = Logger.getLogger(ClientClasspathYamlConfig.class);
+
+    /**
+     * Creates a config which is loaded from a classpath resource using the
+     * Thread.currentThread contextClassLoader. The System.properties are used to resolve variables
+     * in the YAML.
+     *
+     * @param resource the resource, an YAML configuration file from the classpath
+     * @throws IllegalArgumentException              if the resource could not be found
+     * @throws com.hazelcast.core.HazelcastException if the YAML content is invalid
```

Comment:
Minor: can we import `com.hazelcast.core.HazelcastException` so we get tidier Javadoc?

Reply:
Thanks, imported.